torch/linalg/__init__.py

```diff
@@ -140,6 +140,73 @@
     (tensor(3.7417), tensor(11.2250))
 """)
 
+cond = _add_docstr(_linalg.linalg_cond, r"""
+linalg.norm(input, p=None, *, out=None) -> Tensor
+
+Computes the condition number of a matrix :attr:`input`,
+or of each matrix in a batched :attr:`input`, using the matrix norm defined by :attr:`p`.
+The condition number is defined as the matrix norm of
+:attr:`input` times the matrix norm of the inverse of :attr:`input`.
+
+This function supports real-valued, and only on CPU, complex-valued input.
+
+.. note:: For `p = {None, 2, -2}` :attr:`input` can be non-square. For other norm types :attr:`input` must be
+          a square matrix or a batch of square matrices. If :attr:`input` does not satisfy the requirement
+          then a RuntimeError will be thrown.
+
+.. note:: For the batched input if at least one matrix in the batch is not invertible,
+          currently getting the result for all other (possibly) invertible matrices in the batch is not implemented.
+
```

Comment:
A warning about the cross-device data movement needs to appear here, too. Like we did for Cholesky (https://pytorch.org/docs/master/linalg.html?highlight=cholesky#torch.linalg.cholesky). 

However I would suggest we simplify the warning message:

"When given inputs on a CUDA device, this function synchronizes that device with the CPU."

cc @heitorschueroff who will be reviewing these warnings holistically later.